### ISSUE TITLE
Build for minnie: openssl error

### ISSUE THREAD

**Issue by selfthinker** [Maccraft123/Cadmium#18]
When I try to run the build script for minnie I get a fatal error:

```
scripts/extract-cert.c:21:10: fatal error: openssl/bio.h: No such file or directory
   21 | #include <openssl/bio.h>
  	|      	^~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by Maccraft123** [comment]
you are missing build dependencies for linux kernel.
on deb distros add deb-src mirrors to /etc/apt/sources.list and run "sudo apt build-dep linux"

**Comment by selfthinker** [comment]
Thanks for the quick answer.
I'm not very familiar with Linux. Could you spell out which lines to add? (I suspect it's not `deb-src`.) Also, `sudo apt build-dep linux` seems to be missing something, I get a screen with options.

**Comment by Maccraft123** [comment]
What to add really depends on what distro exactly you're using and what do you have in your /etc/apt/sources.list. Can you give a screenshots of screen with options that you get?

**Comment by selfthinker** [comment]
My /etc/apt/sources.list is empty (apart from a couple of comments).

I figured out why I got the screen with options. I had a typo in my command. Now that I've copied the exact command it worked (except it obviously needed some deb-src). Sorry for the false alarm.

My distro is Linux Mint (MATE edition). But I'm happy to set a different distro up if that makes things easier. (I've actually set it up for the sole purpose of building Cadmium.)

**Comment by Maccraft123** [comment]
What I use for developing Cadmium is Debian Sid inside of a chroot.
You don't really need to stick to any distro and I try my best to keep Cadmium as distro-agnostic as possible.

The only requirement probably is just binfmt support, debootstrap, aarch64-linux-gnu OR arm-linux-gnueabihf cross compiler, appropiate headers and a somewhat new qemu user emulation.
I heard from a user that he was able to build Cadmium on a Ubuntu 21.04 system.

**Comment by selfthinker** [comment]
You forgot to mention what to put into /etc/apt/sources.list.
Although I figured it out now. As someone who isn't very familiar with Linux I found everything about sources.list quite confusing. But I found that Linux Mint (and some other distros) have an option in the "Software Sources" app called "Source code repositories". After I enabled that the image finished building successfully.
It might be worth adding something about this to the documentation.
Thanks for your help.
